src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -91,6 +140,49 @@ public static bool CopyFile(string sourceFile, string destinationFile)
             }
         }
 
+        /// <summary>
+        /// Moves source file to destinatino file asynchronously
+        /// </summary>
+        /// <param name="sourceFile"></param>
+        /// <param name="destinationFile"></param>
+        /// <returns></returns>
+        public static async Task<bool> MoveFileAsync(string sourceFile, string destinationFile)
+        {
+            bool result = true;
+
+            if (string.IsNullOrEmpty(sourceFile) || string.IsNullOrEmpty(destinationFile) || !File.Exists(sourceFile))
+            {
```

Comment:
these seem like appropriate cases to throw ArgumentNullException, ArgumentException, or FileNotFoundException

Reply:
I guess we're not doing that elsewhere, so we should be consistent for now.

---
In reply to: [197862456](https://github.com/aspnet/LibraryManager/pull/179#discussion_r197862456) [](ancestors = 197862456)